cli/command_policy_set_test.go

```diff
@@ -0,0 +1,137 @@
+	initialFileFlagVal := *policyIgnoreFileErrors
+	initialDirFlagVal := *policyIgnoreDirectoryErrors
+
+	defer func() {
+		*policyIgnoreFileErrors = initialFileFlagVal
+		*policyIgnoreDirectoryErrors = initialDirFlagVal
+	}()
+
+	for _, tc := range []struct {
+		name           string
+		startingPolicy *policy.ErrorHandlingPolicy
+		fileArg        string
+		dirArg         string
+		expResult      *policy.ErrorHandlingPolicy
+		expErr         bool
+		expChangeCount int
+	}{
+		{
+			name: "No values provided as command line arguments",
+			startingPolicy: &policy.ErrorHandlingPolicy{
+				IgnoreFileErrors:         true,
+				IgnoreFileErrorsSet:      true,
+				IgnoreDirectoryErrors:    true,
+				IgnoreDirectoryErrorsSet: true,
+			},
+			fileArg: "",
+			dirArg:  "",
+			expResult: &policy.ErrorHandlingPolicy{
+				IgnoreFileErrors:         true,
+				IgnoreFileErrorsSet:      true,
+				IgnoreDirectoryErrors:    true,
+				IgnoreDirectoryErrorsSet: true,
+			},
+			expErr:         false,
+			expChangeCount: 0,
+		},
+		{
+			name:           "Malformed arguments",
+			startingPolicy: &policy.ErrorHandlingPolicy{},
+			fileArg:        "not-true-or-false",
+			dirArg:         "not-even-inherit",
+			expResult: &policy.ErrorHandlingPolicy{
+				IgnoreFileErrors:         false,
+				IgnoreFileErrorsSet:      false,
+				IgnoreDirectoryErrors:    false,
+				IgnoreDirectoryErrorsSet: false,
+			},
+			expErr:         true,
+			expChangeCount: 0,
+		},
+		{
+			name:           "One is malformed, the other well formed",
+			startingPolicy: &policy.ErrorHandlingPolicy{},
+			fileArg:        "true",
+			dirArg:         "some-malformed-arg",
+			expResult: &policy.ErrorHandlingPolicy{
+				IgnoreFileErrors:         true,
+				IgnoreFileErrorsSet:      true,
+				IgnoreDirectoryErrors:    false,
+				IgnoreDirectoryErrorsSet: false,
+			},
+			expErr:         true,
+			expChangeCount: 1,
+		},
+		{
+			name: "Inherit case",
+			startingPolicy: &policy.ErrorHandlingPolicy{
+				IgnoreFileErrorsSet:      true,
+				IgnoreDirectoryErrorsSet: true,
+			},
+			fileArg: "inherit",
+			dirArg:  "inherit",
+			expResult: &policy.ErrorHandlingPolicy{
+				IgnoreFileErrors:         false,
+				IgnoreFileErrorsSet:      false,
+				IgnoreDirectoryErrors:    false,
+				IgnoreDirectoryErrorsSet: false,
+			},
+			expErr:         false,
+			expChangeCount: 2,
+		},
+		{
+			name:           "Set to true",
+			startingPolicy: &policy.ErrorHandlingPolicy{},
+			fileArg:        "true",
+			dirArg:         "true",
+			expResult: &policy.ErrorHandlingPolicy{
+				IgnoreFileErrors:         true,
+				IgnoreFileErrorsSet:      true,
+				IgnoreDirectoryErrors:    true,
+				IgnoreDirectoryErrorsSet: true,
+			},
+			expErr:         false,
+			expChangeCount: 2,
+		},
+		{
+			name: "Set to false",
+			startingPolicy: &policy.ErrorHandlingPolicy{
+				IgnoreFileErrors:      true,
+				IgnoreDirectoryErrors: true,
+			},
+			fileArg: "false",
```

Comment:
LG.
It may make sense to add the `("true", "false")` and `("false", "true")` combinations here as well, and at least one with `("inherit", "true")`

Reply:
added